pkg/expression/builtin_json_vec.go

```diff
@@ -860,6 +851,11 @@ func (b *builtinJSONLengthSig) vecEvalInt(ctx EvalContext, input *chunk.Chunk, r
 				return types.ErrInvalidJSONPathMultipleSelection
 			}
 
+			jsonItem := jsonBuf.GetJSON(i)
+			if jsonItem.TypeCode != types.JSONTypeCodeObject && jsonItem.TypeCode != types.JSONTypeCodeArray {
```

Comment:
In non-vector version, if `path` does not exists, it returns null, but in the vector version, if the jsonItem is not JSON object or JSON array, it returns 1 even if `path` does not exists? According to MySQL's behavior, the non-vector version is right.

Reply:
`result.MergeNulls(pathBuf)` has been call in L839, so if `path` is null, `continue` will be call in L843

Reply:
I mean the `path` is not null, but a non exists path. Like `select json_length(cast(1 as json), '$.a')`

Reply:
get! fixed and added test case.